python/dgl/nn/pytorch/gt/lap_enc.py

```diff
@@ -0,0 +1,160 @@
+"""Laplacian Positional Encoder"""
+
+import torch as th
+import torch.nn as nn
+
+
+class LapPosEncoder(nn.Module):
+    r"""Laplacian Positional Encoder (LPE), as introduced in
+    `GraphGPS: General Powerful Scalable Graph Transformers
+    <https://arxiv.org/abs/2205.12454>`__
+
+    This module is a learned laplacian positional encoding module using
+    Transformer or DeepSet.
+
+    Parameters
+    ----------
+    model_type : str
+        Encoder model type for LPE, can only be "Transformer" or "DeepSet".
+    num_layer : int
+        Number of layers in Transformer/DeepSet Encoder.
+    k : int
+        Number of smallest non-trivial eigenvectors.
+    lpe_dim : int
+        Output size of final laplacian encoding.
+    n_head : int, optional
+        Number of heads in Transformer Encoder.
+        Default : 1.
+    batch_norm : bool, optional
+        If True, apply batch normalization on raw LaplacianPE.
+        Default : False.
+    num_post_layer : int, optional
+        If num_post_layer > 0, apply an MLP of ``num_post_layer`` layers after
+        pooling.
+        Default : 0.
+
+    Example
+    -------
+    >>> import dgl
+    >>> from dgl import LaplacianPE
+    >>> from dgl.nn import LapPosEncoder
+
+    >>> transform = LaplacianPE(k=5, feat_name='eigvec', eigval_name='eigval', padding=True)
+    >>> g = dgl.graph(([0,1,2,3,4,2,3,1,4,0], [2,3,1,4,0,0,1,2,3,4]))
+    >>> g = transform(g)
+    >>> eig_vals, eig_vecs = g.ndata['eigval'], g.ndata['eigvec']
+    >>> TransformerLPE = LapPosEncoder(model_type="Transformer", num_layer=3, k=5,
+                                       lpe_dim=16, n_head=4)
+    >>> pos_enc = TransformerLPE(eig_vals, eig_vecs)
+    >>> DeepSetLPE = LapPosEncoder(model_type="DeepSet", num_layer=3, k=5,
+                                   lpe_dim=16, num_post_layer=2)
+    >>> pos_enc = DeepSetLPE(eig_vals, eig_vecs)
+    """
+
+    def __init__(
+        self,
+        model_type,
+        num_layer,
+        k,
+        lpe_dim,
+        n_head=1,
+        batch_norm=False,
+        num_post_layer=0,
+    ):
+        super(LapPosEncoder, self).__init__()
+        self.model_type = model_type
+        self.linear = nn.Linear(2, lpe_dim)
+
+        if self.model_type == "Transformer":
+            encoder_layer = nn.TransformerEncoderLayer(
+                d_model=lpe_dim, nhead=n_head, batch_first=True
+            )
+            self.pe_encoder = nn.TransformerEncoder(
```

Comment:
what does pe in pe_encoder stand for?

Reply:
pe stands for positional encoding, which is a common abbr. in this context.